vac/raw/consensus-hashgraphlike.md

```diff
@@ -0,0 +1,212 @@
+to the random peer from the network or sends it to the proposal to the specific channel.
+
+## 2. Exchanging votes across the peers
+
+Once the peer receives the proposal message P_1 from a 1-1 or a gossipsub channel with its votes does the following checks:
+
+- Check the signatures of the voting(should it be the recursive from the first one to the last one)
+- Check the hash checks in particular:
+  - Parenthash check prevents double voting
+  - Receivedhash check provides tampering attacks
+
+In particular, proposal message P_1 where its vote V_1 = P_1.votes[0] and performs as follows:
+
+- Verifiy the vote V_1.signature with with the V_1.vote_owner in the proposal.
+- If the peer verifies the signature,
+it continues to create P_2 with the new vote V_2 that consists of as following:
+  - adding its public key
+  - timestamp
+  - boolean vote
+  - V_2.parent_hash = 0 if there is no previous peer's vote, otherwise Hash of previous owner's vote
+  - V_2.received_hash = P_1.votes[0]
+  - Calculate vote_hash by  hash of Vote hash(vote_id, owner, timestamp, vote, parent_hash, received_hash)
+    then adds the V_2.vote_hash
+  - Sign vote_hash with its private key then adds V_2.vote_hash.
+- Create P_2 with by adding V_2 as follows:
+  - P_2.name, P_2.proposal_id and P_2.proposal_owner are the same with P_1.
+  - Add the V_2 to the P_2.Votes list.
+  - Increase the round by one, namely P_2.round = P_1.round + 1.
+  - Verify the time proposal timestamp is valid for expiration time, namely P_2.timestamp - current < P_1.expiration_time.
+  If this does not hold, other peers ignore the message.
+
+After the peer creates the proposal P_2 with its vote V_2,
+sends it to the random peer from the network or
+sends it to the proposal to the specific channel.
+
+## 3. Determining the Result
+
+Since a peer cannot know if the votes are enough for the consensus without counting,
+it requires checking each time by the peers.
+The voting result is set YES if the majority of the 2n/3 from the distinct peers vote YES.
+
+To check the findDistintVoter method takes the proposal and extracts the number of distinct voters by
+traversing the Votes list in the proposal.
+If this method returns true, then we check the strong validation of the vote as follows:
+
+- Check each signature in the vote
+  - as shown in the section Exchanging votes across the peers.
+- Check the parent hash chain
+  - if there are multiple votes from the same owner namely vote_i and vote_i+1 respectively,
+    the parent hash of vote_i+1 should be the hash of vote_i
+- Check the previous hash chain
+  - each received hash of vote_i+1 should be equal to the hash of vote_i.
+- Check the timestamp against the replay attack:
+  - timestamps check the freshness of the message against the replay.
+    In particular, the timestamp cannot be the old in the determined threshold.
+
+If a proposal is verified by all the checks,
+the countVote method counts each YES vote from the list of Votes.

```

Comment:
looks like we can make a list with API, you already mentioned some of them, what do you think?